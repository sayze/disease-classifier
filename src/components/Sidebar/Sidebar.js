import React from 'react'
import { Paper, List, ListItem, ListItemIcon, ListItemText, ListSubheader } from '@material-ui/core'
import { FavoriteRounded, EcoSharp } from '@material-ui/icons'
import PropTypes from 'prop-types'
import { menu } from '.'

const Sidebar = ({ activeItem, onItemChange }) => (
  <Paper variant="outlined" style={{ height: '100%' }}>
    <List component="ul" aria-label="main mailbox folders">
      <ListSubheader>Perform Classification</ListSubheader>
      <ListItem button selected={activeItem === menu.HYPERTENSION} onClick={() => onItemChange(menu.HYPERTENSION)}>
        <ListItemIcon>
          <FavoriteRounded />
        </ListItemIcon>
        <ListItemText primary="Hypertension" />
      </ListItem>
      <ListItem button selected={activeItem === menu.KIDNEY} onClick={() => onItemChange(menu.KIDNEY)}>
        <ListItemIcon>
          <EcoSharp />
        </ListItemIcon>
        <ListItemText primary="Kidney Disease" />
      </ListItem>
    </List>
  </Paper>
)

Sidebar.defaultProps = {
  onItemChange: () => {},
}

Sidebar.propTypes = {
  activeItem: PropTypes.string.isRequired,
  onItemChange: PropTypes.func,
}

export default Sidebar
