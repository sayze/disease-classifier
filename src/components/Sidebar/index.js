/**
 * Object containing constants that can be consistently used across multiple components/pages.
 */
export const Classifications = {
  HYPERTENSION: 'hypertension',
  KIDNEY: 'kidney',
}

export { default } from './Sidebar'
