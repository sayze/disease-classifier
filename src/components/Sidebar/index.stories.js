import React from 'react'
import { withKnobs, select } from '@storybook/addon-knobs'
import Sidebar, { Classifications } from '.'

export default {
  title: 'Sidebar',
  component: Sidebar,
  decorators: [withKnobs],
}

// Setup variables for select knob.
const options = [Classifications.HYPERTENSION, Classifications.KIDNEY]
const defaultValue = Classifications.HYPERTENSION

export const Default = () => <Sidebar activeItem={select('Acive Item', options, defaultValue)} />
