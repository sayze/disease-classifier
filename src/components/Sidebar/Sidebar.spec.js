import React from 'react'
import { shallow } from 'enzyme'
import Sidebar, { Classifications } from '.'

describe('Sidebar', () => {
  it('should render Sidebar with hypertension active', () => {
    const component = shallow(<Sidebar activeItem={Classifications.HYPERTENSION} />)
    expect(component).toMatchSnapshot()
  })
})
