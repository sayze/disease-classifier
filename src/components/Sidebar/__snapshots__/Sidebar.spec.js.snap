// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render Sidebar with hypertension active 1`] = `
<WithStyles(ForwardRef(Paper))
  variant="outlined"
>
  <WithStyles(ForwardRef(List))
    component="ul"
  >
    <WithStyles(ForwardRef(ListSubheader))>
      Perform Classification
    </WithStyles(ForwardRef(ListSubheader))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      onClick={[Function]}
      selected={true}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <FavoriteRoundedIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Hypertension"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      onClick={[Function]}
      selected={false}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <EcoSharpIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Kidney Disease"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
</WithStyles(ForwardRef(Paper))>
`;
